inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -616,14 +622,57 @@ public void deleteProperty(KnowledgeBase kb, KBProperty aType)
             tupleQuery.setBinding("pTYPE", aKB.getTypeIri());
             tupleQuery.setBinding("oPROPERTY", aKB.getPropertyTypeIri());
             tupleQuery.setBinding("pLABEL", aKB.getPropertyLabelIri());
-            tupleQuery.setBinding("pDESCRIPTION", aKB.getPropertyDescriptionIri());
+            tupleQuery.setBinding("pDESCRIPTION", aKB.getDescriptionIri());
+            tupleQuery.setBinding("pSUBPROPERTY", aKB.getSubPropertyIri());
             tupleQuery.setIncludeInferred(aIncludeInferred);
             return evaluateListQuery(aKB, tupleQuery, false, aAll, "s");
         });
         resultList.sort(Comparator.comparing(KBObject::getUiLabel));
         return resultList;
     }
+    
+    @Override
+    public Set<KBHandle> getSubPropertyLabels(KnowledgeBase aKB)
+    {
+        Set<KBHandle> subPropertyLabels = new HashSet<KBHandle>();
+        KBHandle labelHandle = new KBHandle(aKB.getLabelIri().stringValue());
+        getIterativeSubPropertyLabels(subPropertyLabels, aKB,
+                labelHandle);
+        return subPropertyLabels;
+       
+    }
+    
+    public Set<KBHandle> getIterativeSubPropertyLabels(Set<KBHandle> subPropertyLabelSet,
```

Comment:
You mean `getSubPropertyLabelsRecursively`?

Reply:
Oh yeah.. Sorry . Will change that.